Generals/Code/GameEngineDevice/Source/W3DDevice/GameClient/W3DDisplay.cpp

```diff
@@ -2756,6 +2757,43 @@ VideoBuffer*	W3DDisplay::createVideoBuffer( void )
 	return buffer;
 }
 
+//============================================================================
+// W3DDisplay::drawScaledVideoBuffer
+//============================================================================
+
+void W3DDisplay::drawScaledVideoBuffer()
+{
+	// TheSuperHackers @bugfix Mauller 20/07/2025 scale videos based on screen size so they are shown in their original aspect
+	float videoAspect = (float)m_videoStream->width() / (float)m_videoStream->height();
+	float displayAspect = (float)getWidth() / (float)getHeight();
+	bool wideaspect = displayAspect >= videoAspect;
+
+	int frameStartWidth = 0;
+	int frameEndWidth = 0;
+	int frameStartHeight = 0;
+	int frameEndHeight = 0;
```

Comment:
Maybe call these startX, startY, endX, endY to be consistent with the names in drawVideoBuffer.